proposals/action-on-user-settings-changed-api.md

```diff
@@ -0,0 +1,68 @@
+# Extension API Proposal: action.onUserSettingsChanged
+
+## Background
+
+Proposal based on [issue #346](https://github.com/w3c/webextensions/issues/346): New API: UserSettings.isOnToolbar change event
+
+## API Overview
+
+### Objective
+
+Fire an event when user-specified settings relating to an extension's action changed.
+
+### Use Cases
+
+#### Pinning / unpinning extension
+
+Extensions can know if its action is pinned by calling `chrome.action.getUserSettings.` If the extension wants to know when this value changes, it has to poll the API method. Instead, it could register for an event.
+
+## Implementation
+
+### API Schema
+
+#### Types
+
+```javascript
+dictionary UserSettings {
+  isOnToolbar: boolean
+}
+```
+
+#### Events
+
+```javascript
+// Fires when user-specified settings relating to an extension's action changed.
+<browser>.action.onUserSettingsChanged.addListener(
+  callback: function, // where callback looks like: (userSettings: UserSettings) => void
```

Comment:
The type of the event parameter is identical to the return value of `getUserSettings`. Is that an intentional design?

With the current proposal consisting of one property only, it does not make that much of a difference. Once multiple events get added, the answer depends on the design decision:

- If we want to always include the latest value of all settings, then we can use the same type as the event parameter to `onUserSettingsChanged` listeners and as the return value of `getUserSettings`.
- If we want to only include modified values, then we need a new type that looks like `onUserSettingsChanged`, but with all fields optional.

Reply:
Great catch! We think it's more valuable to the user to only return the modified values. Updated the proposal